server.py

```diff
@@ -1,18 +1,145 @@
-# _*_ coding utf-8 _*_
+"""Simple HTTP server."""
 import socket
+import io
+import os
 
+address = ('127.0.0.1', 5003)
 server = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
-
-address = ('127.0.0.1', 5000)
 server.bind(address)
 server.listen(1)
-conn, addr = server.accept()
+buffer_length = 1024
+server_path = "/Users/admin-1/http_server/http-server/webroot"
+
+
+def server_func():
+    """Run the server."""
+    try:
+        while True:
+            conn, addr = server.accept()
+            try:
+                while True:
+                    data = conn.recv(buffer_length)
+                    if len(data) == buffer_length:
+                        # Im not positive this if statement is needed, but everything is
+                        # working so it's staying for now
+                        pass
+                    elif len(data) < buffer_length:
+                        # full request has been received
+                        try:
+                            # check for uri request for errors
+                            uri = parse_request(data)
+                            # generate response based on file type
+                            new_uri = route_uri(server_path, uri)
+                            # send response to browser/client
+                            conn.sendall(new_uri)
+                        except AttributeError:
+                            # uri missing data
+                            missing_error = "HTTP/1.1 400 Bad Request\r\n"
+                            conn.sendall(missing_error.decode('utf-8'))
+                        except NameError:
+                            # uri has unsupported method
+                            method_error = "HTTP/1.1 405 Method Not Allowed\r\n"
+                            conn.sendall(method_error.decode('utf-8'))
+                        except TypeError:
+                            # uri has http version other than 1.1
+                            version_error = "HTTP/1.1 505 Version Not Supported\r\n"
+                            conn.sendall(version_error.decode('utf-8'))
+                        conn.close()
+            except:
+                conn.close()
+    except KeyboardInterrupt:
+        server.close()
+
+
+def parse_request(request):
+    """Make sure client request is decent."""
+    temp_list = request.split("\r\n")
+    part_list = temp_list[0].split(' ')
+    if len(part_list) < 3:
+        raise AttributeError
+    elif part_list[0] != 'GET':
+        raise NameError
+    elif part_list[2] != "HTTP/1.1":
+        raise TypeError
+    else:
+        return part_list[1]
+
+
+def response_ok(path, uri, file_type):
+    """Create customized HTTP response for route_uri()."""
+    return_path = os.path.join(path, uri)
+    if os.path.isfile(return_path) is False:
+        return four_oh_four()
+    else:
+        if "image" in file_type:
+            open_file = io.open(return_path, 'rb')
```

Comment:
What if there were a PDF file in the server directory?  Is that bytes or unicode?